core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolutionHelperService.js

```diff
@@ -0,0 +1,76 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for solution editor.
+ */
+
+oppia.constant('SUPPORTED_INTERACTIONS', [
+    'InteractiveMap',
+    'MusicNotesInput',
+    'GraphInput',
+    'SetInput',
+    'MathExpressionInput',
+    'MultipleChoiceInput',
+    'ImageClickInput',
+    'ItemSelectionInput'
+]);
+
+oppia.constant('UNSUPPORTED_INTERACTION_OBJECT_TYPES', {
```

Comment:
I think some comments are needed here to explain what exactly "supported interaction" and "unsupported interaction" mean. I also think that the naming could be more specific, since these are globally-visible constants. It isn't that Oppia as a whole does not support the interaction; it's more like some part of Oppia has a problem with it...

I think it's also worth explaining why some interactions are not supported and whether there's something that can be done to support them, or whether they're fundamentally unsupportable.